src/@ionic-native/plugins/fcm/index.ts

```diff
@@ -0,0 +1,171 @@
+import { Plugin, IonicNativePlugin, Cordova } from '@ionic-native/core';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+export interface NotificationData {
+
+  /**
+   * Determines whether the notification was pressed or not
+   */
+
+  wasTapped: boolean;
+
+  /**
+   * Notification data hash item
+   */
+
+  [name: string]: any;
+
+}
+
+/**
+ * @name FCM
+ * @description
+ * Provides basic functionality for Firebase Cloud Messaging
+ *
+ * @usage
+ * ```typescript
+ * import { FCM } from 'ionic-native';
+ *
+ * constructor(private fcm: FCM) {}
+ * 
+ * ...
+ * 
+ * fcm.subscribeToTopic('marketing');
+ * 
+ * fcm.getToken().then(token=>{
+ *   backend.registerToken(token);
+ * })
+ * 
+ * fcm.onNotification().subscribe(data=>{
+ *   if(data.wasPressed){
+ *     console.log("Received in background");
+ *   } else {
+ *     console.log("Received in foreground");
+ *   };
+ * })
+ * 
+ * fcm.onTokenRefresh().subscribe(token=>{
+ *   backend.registerToken(token);
+ * })
+ * 
+ * fcm.unsubscribeToTopic('marketing');
+ * 
+ * ```
+ * @interfaces
+ * NotificationData
+ */
+@Plugin({
+  pluginName: 'FCM',
+  plugin: 'cordova-plugin-fcm',
+  pluginRef: 'FCMPlugin',
+  repo: 'https://github.com/fechanique/cordova-plugin-fcm',
+  platforms: ['iOS', 'Android']
+})
+@Injectable()
+export class FCM extends IonicNativePlugin {
+
+  /**
+   * Get's device's current registration id
+   * 
+   * ```typescript
+   * 
+   * fcm.getToken().then(token=>{
+   *   backend.registerToken(token)
+   * })
+   * 
+   * ```
+   * 
+   * @returns {Promise<string>} Returns a Promise that resolves with the registration id token
+   */
+  @Cordova()
+  getToken(): Promise<string> { return; }
+
+  /**
+   * Event firing on the token refresh
+   * 
+   * ```typescript
+   * 
+   * let tokenChange = fcm.onTokenRefresh().subscribe(token=>{
+   *   backend.registerToken(token)
+   * })
+   * 
+   * // To unsubscribe from changes 
+   * 
+   * tokenChange.unsubscribe()
+   * 
+   * ```
+   * 
+   * @returns {Observable<string>} Returns an Observable that notifies with the change of device's registration id
+   */
+  @Cordova({
+    observable: true,
+    successIndex: 0
+  })
+  onTokenRefresh(): Observable<string> { return; }
+
+  /**
+   * Subscribes you to a [topic](https://firebase.google.com/docs/notifications/android/console-topics)
+   * 
+   * ```typescript
+   * 
+   * fcm.subscribeToTopic('marketing');
+   * 
+   * ```
+   * 
+   * @param {string} topic Topic to be subscribed to
+   * 
+   * @returns {Promise<any>} Returns a promise resolving in result of subscribing to a topic
+   */
+  @Cordova({
+    callbackOrder: 'reverse'
```

Comment:
This method should not have `callbackOrder: 'reverse'`. See https://github.com/fechanique/cordova-plugin-fcm/blob/master/www/FCMPlugin.js#L8